clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3558,14 +3578,20 @@ class SYCLKernelNameTypeVisitor
 
 void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
                                ArrayRef<const Expr *> Args) {
+  QualType KernelNameType =
```

Comment:
This means the kernel name type check is now done for device and host compile right? I assume that means this results in multiple diagnostics when doing and end-to-end run right?

Reply:
I also had this thought so I checked that. The diagnostic is emitted only once for any error (connected to incorrect name or not). I was curious why, but I didn't think it is a good idea to spend time and debug it. Okay, for incorrect kernel names all diagnostics (not only the new one) are actually emitted twice because SYCL headers declare second kernel for range rounding optimization, but the compiler sees it as a separate kernel so its output is correct.
I remember times when ALL errors were emitted twice because of two passes of FE, I believe it was just fixed somehow.

Reply:
> This is existing behavior right?

Yes. I checked that for other kernel name diagnostics (like kernel name cannot be in std namespace).